### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by AnsarIbrahim**
### This PR contains :
 
- Set up React Testing Library✅
- Write unit tests for the files operate.js and calculate.js using Jest.✅
- Create unit tests for all React components:✅
- Use Jest snapshots to test the components.✅
- Use React Testing Library to simulate user interaction.✅
- Run your full tests suite using Jest. All tests should pass✅

### CODE REVIEW

File: src/test/calculate.test.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import calculate from '../logic/calculate';
+
+describe('calculate', () => {
+  it('clears the calculator data when "AC" button is pressed', () => {
+    const result = calculate({ total: '10', next: '5', operation: '+' }, 'AC');
+    expect(result).toEqual({ total: null, next: null, operation: null });
+  });
+
+  it('updates the next number when a number button is pressed', () => {
+    const result = calculate({ total: '10', next: '5', operation: '+' }, '7');
```
Comment (AdnanOlarmmi) [inline src/test/calculate.test.js:10]: - [ ] Nice one with the tests, according to the requirement, you're supposed to write tests for all scenarios of the calculate.js and the operate.js, the operate.js seems okay, but the following was omitted in the calculate.js

- [ ] You didn't test if the input has a decimal point.

```
 if (buttonName === '.') {
    if (obj.next) {
      if (obj.next.includes('.')) {
        return { ...obj };
      }
      return { ...obj, next: `${obj.next}.` };
    }
    if (obj.operation) {
      return { ...obj, next: '0.' };
    }
    if (obj.total) {
      if (obj.total.includes('.')) {
        return {};
      }
      return { ...obj, next: `${obj.total}.` };
    }
    return { ...obj, next: '0.' };
  }
```

File: src/test/calculate.test.js
Comment (AdnanOlarmmi) [inline src/test/calculate.test.js:1]: - [ ] Nice work with the following tests, but according to the project requirement, the test should also simulate user's interaction.

**Review by AdnanOlarmmi — CHANGES_REQUESTED** [review]
Hi @AnsarIbrahim and @Innocent259, 

This is Adnan Is-haq, your beloved code reviewer,

You've done a very good job so far, but a few things still need to be worked on before moving on to the next project, but you're so close!

You're Almost there, funny one eyed alien

## Highlights
- Well written test, Kudos to you.
- Nice working in pairs, this shows your collaborative skills.

## Required Changes ♻️
- [ ] Please, kindly include your partner's information in the authors section of your readme.
image

_Check the comments under the review._

## Optional suggestions
- N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by KayLemba — APPROVED** [review]
# STATUS: APPROVED :green_circle: :clap: 

Hello @TEAM  , 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉  Keep up the good work 🥂 

HIGHLIGHTS:

- [x]  Professional README
- [x] No linter errors
- [x] Descriptive P.R
- [x] Tests are running as intended 🥇  🥂 



 
 

 










 








Now you can merge this branch and move on :1st_place_medal: 
Happy Coding cheers 🥇 💯 



well-1

Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.



Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._]]

**Pull request merged**